drivers/hmis/app/graphql/mutations/clean_base_mutation.rb

```diff
@@ -0,0 +1,20 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# avoid pollution from relay
```

Comment:
Can you say a little more about which kinds of mutations would warrant using Relay vs not, if you know? I don't really know anything about Relay (or GraphQL internals..) but it seems like we use it for pagination and it claims to provide some benefits there.

Reply:
Relay is a set of conventions for behavior in graphql; IIRC it was intended address FB's use cases. I think our pagination is offset-based (from what I've seen), my understanding is that Relay convention is cursor-based. 

I could be missing something but I don't really understand the value of using the relay behaviors; there are frameworks or clients that expect these behaviors but they don't seem relevant to us

For mutations specifically:
https://graphql-ruby.org/api-doc/2.0.21/GraphQL/Schema/RelayClassicMutation

> An argument called clientMutationId is always added, but it’s not passed to the resolve method. The value is re-inserted to the response. (It’s for client libraries to manage optimistic updates.)

Does this seem like something we need? For this app I imagine optimistic updates aren't what we'd want. Or if they are, it's probably just a few mutations

> The mutation accepts one argument called input, arguments defined in the mutation class are added to that input object, which is generated by the mutation.

I think using input types is great but this wraps every mutation up in an automatically created "mutation#{Input}" class. This means that:
* clutters schema and TS types
* we lose the ability to just pass an ID or something simple to a mutation directly
* additional complexity due to interaction with Relay (for example trying to define an input class with the same name as the automatically generated one)

Reply:
Thank you for explaining all that! I will do some more reading up about it. My initial reaction is:
* We don't do cursor-based pagination, and we don't do optimistic updates
* Our type files on the frontend are definitely huge
* I always found the automatic `input` wrapper bothersome, but I figured that was just a fact of life with ruby-graphql

I see that HNMI uses Relay, which might be why Zach set us up to use it too. @ZachCMP is that right, or was it an intentional decision?. Also it looks like you set up Relay in HNMI in the first place, @ttoomey , so maybe you had some reason to do so. (I won't expect you to remembering the reason 3 years later though 😄 )

Reply:
@gigxz IIRC the Ruby GraphQL setup uses those relay features by default. I didn't go out of my way to set them up.

Reply:
Oh I just realized that Sandy did the setup for Relay (the edges and connections and whatnot - I have no idea what they're for) so I'll ask him about that.

Reply:
Yeah I think you get Relay when you ask graphql-ruby to scaffold out the base classes 🤦